internal/server/routes.go

```diff
@@ -250,6 +251,9 @@ func wrapRoute[Req, Res any](a *API, routeID routeIdentifier, route route[Req, R
 			Response:      &access.ResponseMetadata{},
 		}
 		c.Set(access.RequestContextKey, rCtx)
+		if a.server.Google != nil {
+			c.Set(data.GoogleProviderContextKey, a.server.Google)
```

Comment:
We need a way to include some `Server` data on `RequestContext`. I've been thinking a new field like `Server` would allow us to do that. That would remove the need to set anything on `gin.Context`. The API handler for `ListProviders` could read the field directly.

Here we would simply set `Server: a.server`.

Reply:
Update to not use the context after our discussion